06.bowling_object/game.rb

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require './shot'
+require './frame'
+
+class Game
+  def initialize(score)
+    @frames = parse_frames(score)
+  end
+
+  def parse_frames(score)
+    frames_mark = score.split(',')
+    frames = []
+    9.times do
+      frames << if frames_mark[0] == 'X'
+                  Frame.new(frames_mark.shift)
+                else
+                  Frame.new(frames_mark.shift, frames_mark.shift)
+                end
+    end
+    frames << Frame.new(frames_mark[0], frames_mark[1], frames_mark[2])
+    frames
+  end
+
+  def calc
+    total_score = 0
+    @frames.each_with_index do |frame, count|
+      total_score += frame.score
+      next if frame.score != 10
+
+      total_score += if frame.first_shot.mark == 'X'
+                       strike(count)
+                     else
+                       spare(count)
+                     end
```

Comment:
```suggestion
      total_score += frame.score(何らかの引数)
```

のようにしてframe自身がstrikeやspareのボーナス点も算出できるようにした方がオブジェクト指向プログラミングらしいかもしれません。
（今の設計だとGameクラスにロジックが集中しすぎてしまっている）

Reply:
frameクラスにボーナス点を算出するメソッドを実装しました